### PR TITLE
feat: [CDS-52600]: Deploy step handler with s3 artifact source

### PR DESCRIPTION
**Pull request by rohitk-h**
## Describe your changes
- The PR contains changes for aws lambda deploy step. User can create and update aws lambda native function using this step. 
- The artifact can be stored in a s3 bucket. 
- The step is behind feature flag

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-ALL: `trigger tiAll`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44069)

### CODE REVIEW

**Review by sainath-b — APPROVED** [review]
> Reviewed 23 of 34 files at r1, 11 of 11 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44069)* status: 34 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by sainath-b — CHANGES_REQUESTED** [review]
@rohitk-h UTs are failing. Please fix.

**Review by PragyeshMishra01 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44069)* status: 34 file(s) to review. 1 change requests. 0 of 2 approvals obtained. Waiting on team(s): del-team. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 272 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOjeR00-1Dus-UyYAOx:-NOjeR00-1Dus-UyYAOy:b-81v7f0) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L272)):*
> ```Java
> 
>     ManifestOutcome awsLambdaManifestOutcome = getAwsLambdaManifestOutcome(manifestsOutcome.values());
> ```

are we not supporting harness store for manifest
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 325 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOjdnHZ0MH6EGgr3urH:-NOjdnHZ0MH6EGgr3urI:b-33cs7c) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L325)):*
> ```Java
>     StoreConfig storeConfig = manifestOutcome.getStore();
>     if (!ManifestStoreType.isInGitSubset(storeConfig.getKind())) {
>       throw new InvalidRequestException("Invalid kind of storeConfig for Ecs step", USER);
> ```

should not be Ecs here in message exception
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaCommandRequest.java` line 61 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOjh5m-0ZdzXr3RCGyw:-NOjh5m-0ZdzXr3RCGyx:bv9e2fw) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaCommandRequest.java#L61)):*
> ```Java
>     if (awsLambdaArtifactConfig instanceof AwsLambdaS3ArtifactConfig) {
>       AwsConnectorDTO connectorConfigDTO = (AwsConnectorDTO) ((AwsLambdaS3ArtifactConfig) awsLambdaArtifactConfig)
>                                                .getConnectorDTO()
> ```

nit: We should add NPE check for awsLambdaArtifactConfig, if it is null for prepareRollbackRequest
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaPrepareRollbackRequest.java` line 45 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOjfwWF43ka31YYkX0O:-NOjfwWF43ka31YYkX0P:b-j0d31j) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaPrepareRollbackRequest.java#L45)):*
> ```Java
>   public AwsLambdaInfraConfig getAwsLambdaInfraConfig() {
>     return null;
>   }
> ```

is this intentional?

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java [hunk 1]
```diff
@@ -92,4 +149,266 @@ public ManifestOutcome getAwsLambdaManifestOutcome(@NotEmpty Collection<Manifest
...
+
+    return getGitFetchFileTaskResponse(
+        ambiance, false, stepElementParameters, awsLambdaStepPassThroughData, gitRequestFileConfig);
+  }
+
+  private GitRequestFileConfig getGitFetchFilesConfigFromManifestOutcome(
+      ManifestOutcome manifestOutcome, Ambiance ambiance) {
+    StoreConfig storeConfig = manifestOutcome.getStore();
+    if (!ManifestStoreType.isInGitSubset(storeConfig.getKind())) {
+      throw new InvalidRequestException("Invalid kind of storeConfig for Ecs step", USER);
```
Comment (sainath-b) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java:325]: Remove Ecs

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTask.java [hunk 2]
```diff
@@ -9,21 +9,33 @@
...
+import com.google.inject.Inject;
+import java.util.Map;
 import java.util.function.BooleanSupplier;
 import java.util.function.Consumer;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.NotImplementedException;
-
+@Slf4j
 @OwnedBy(HarnessTeam.CDP)
 public class AwsLambdaCommandTask extends AbstractDelegateRunnableTask {
```
Comment (sainath-b) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTask.java:34]: @rohitk-h We shall use separate delegate task handler for each type of aws lambda task for backward compatibility

**Review by sainath-b — CHANGES_REQUESTED** [review]
@rohitk-h Please address comments

File: 260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java [hunk 3]
```diff
@@ -2150,6 +2152,9 @@ private void bindDelegateTasks() {
         .toInstance(GoogleFunctionPrepareRollbackTask.class);
     mapBinder.addBinding(TaskType.GOOGLE_FUNCTION_ROLLBACK_TASK).toInstance(GoogleFunctionRollbackTask.class);
     mapBinder.addBinding(TaskType.GOOGLE_FUNCTION_TRAFFIC_SHIFT_TASK).toInstance(GoogleFunctionTrafficShiftTask.class);
+
+    // AWS Lambda
+    mapBinder.addBinding(TaskType.AWS_LAMBDA_DEPLOY_COMMAND_TASK_NG).toInstance(AwsLambdaCommandTask.class);
```
Comment (piyush-bhuwalka) [inline 260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java:2157]: we should have independent task types for each step

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaDeployRequest.java [hunk 4]
```diff
@@ -28,8 +29,10 @@ public class AwsLambdaDeployRequest implements AwsLambdaCommandRequest, NestedAn
   AwsLambdaCommandTypeNG awsLambdaCommandTypeNG;
   String commandName;
   CommandUnitsProgress commandUnitsProgress;
-  @NonFinal @Expression(ALLOW_SECRETS) AwsLambdaFunctionsInfraConfig awsLambdaFunctionsInfraConfig;
+  @NonFinal @Expression(ALLOW_SECRETS) Integer timeoutIntervalInMin;
+  @NonFinal @Expression(ALLOW_SECRETS) AwsLambdaInfraConfig awsLambdaInfraConfig;
   @NonFinal @Expression(ALLOW_SECRETS) String awsLambdaDeployManifestContent;
+  @NonFinal @Expression(ALLOW_SECRETS) AwsLambdaArtifactConfig awsLambdaArtifactConfig;
   String functionName;
```
Comment (piyush-bhuwalka) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaDeployRequest.java:36]: @NonFinal @Expression(ALLOW_SECRETS) for functionName and qualifier as well

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambda.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.Value;
+
+@Value
+@Builder
+@OwnedBy(HarnessTeam.CDP)
+public class AwsLambda {
+  String functionName;
+  String runtime;
+  String functionArn;
+  String version;
```
Comment (piyush-bhuwalka) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambda.java:23]: there should be tags , aliases and other fields as well. We can look into GetFunction Response DTO for the fields

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/beans/AwsLambdaStepOutcome.java [hunk 6]
```diff
@@ -24,8 +24,9 @@
 @TypeAlias("awslambdaStepOutcome")
 @JsonTypeName("awslambdaStepOutcome")
 @RecasterAlias("io.harness.cdng.aws.lambda.beans.AwsLambdaStepOutcome")
-
 public class AwsLambdaStepOutcome implements Outcome, ExecutionSweepingOutput {
   String functionName;
   String runtime;
+  String version;
+  String functionArn;
 }
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/beans/AwsLambdaStepOutcome.java:32]: Lets add AwsLambdaPrepareRollbackOutcome DTO as well in this PR

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java [hunk 7]
```diff
@@ -92,4 +149,266 @@ public ManifestOutcome getAwsLambdaManifestOutcome(@NotEmpty Collection<Manifest
...
+    // Add manifestContent
+    AwsLambdaStepPassThroughData awsLambdaStepPassThroughDataWithManifestContent =
+        AwsLambdaStepPassThroughData.builder()
+            .manifestContent(manifestContent)
+            .manifestsOutcome(awsLambdaStepPassThroughData.getManifestsOutcome())
+            .infrastructureOutcome(awsLambdaStepPassThroughData.getInfrastructureOutcome())
+            .unitProgressData(awsLambdaStepPassThroughData.getUnitProgressData())
+            .build();
+
+    return executeTask(ambiance, stepParameters, awsLambdaStepPassThroughDataWithManifestContent);
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java:202]: assuming prepare rollback subtask will be part of another PR

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java [hunk 8]
```diff
@@ -50,7 +100,14 @@ public class AwsLambdaHelper extends CDStepHelper {
...
 
-  public AwsLambdaFunctionsInfraConfig getInfraConfig(InfrastructureOutcome infrastructure, Ambiance ambiance) {
+  @Inject private EntityReferenceExtractorUtils entityReferenceExtractorUtils;
+
+  @Inject private PipelineRbacHelper pipelineRbacHelper;
+
+  private final String AWS_LAMBDA_PREPARE_ROLLBACK_COMMAND_NAME = "PrepareRollbackAwsLambda";
+  private final String AWS_LAMBDA_DEPLOY_COMMAND_NAME = "DeployAwsLambda";
+
+  private AwsLambdaFunctionsInfraConfig getInfraConfig(InfrastructureOutcome infrastructure, Ambiance ambiance) {
```
Comment (piyush-bhuwalka) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java:110]: make this public. This can be used in several steps

**Review by rohitk-h — COMMENTED** [review]
UT's fixed, comments addressed. 

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44069#-:-NOmXvI1DFj_IXrFjajH:bd93d81)* status: 34 file(s) to review. 1 change requests. 13. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): del-team. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 110 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOTCEJ2JicewIFx0Ptg-r2-110:-NOmSpywEhciv4FdIsIu:boac9ai) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L110)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

Lets make this public. This can be used in several steps
</blockquote></details>

updated, 
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 202 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOTCEJ2JicewIFx0Ptg-r2-202:-NOmSr_61bpAeVgeFRSB:b-7g7nm2) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L202)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

assuming prepare rollback subtask will be part of another PR 
</blockquote></details>

yes, 
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 272 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOjeR00-1Dus-UyYAOx:-NOmSRk_Cn3ZajFdwcUN:b-3t3qa7) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L272)):*
<details><summary><i>Previously, PragyeshMishra01 (Pragyesh Mishra) wrote…</i></summary><blockquote>

are we not supporting harness store for manifest
</blockquote></details>

not in the first iteration, we'll implement in phase 2. 
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 325 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOjdnHZ0MH6EGgr3urH:-NOmSsry1UTanpTaMh-r:bvnrpcm) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L325)):*
<details><summary><i>Previously, PragyeshMishra01 (Pragyesh Mishra) wrote…</i></summary><blockquote>

should not be Ecs here in message exception
</blockquote></details>

updated
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 325 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOTCEJ2JicewIFx0Ptg-r2-325:-NOmSxQmBgaPEg95S0uk:bvnrpcm) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L325)):*
<details><summary><i>Previously, sainath-b (Sainath Batthala) wrote…</i></summary><blockquote>

Remove Ecs
</blockquote></details>

updated
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/beans/AwsLambdaStepOutcome.java` line 32 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOTCEJ2JicewIFx0Ptk-r2-32:-NOmTSIz3IZPWP1lsjr5:b-7tfmjp) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/beans/AwsLambdaStepOutcome.java#L32)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

Lets add AwsLambdaPrepareRollbackOutcome DTO as well in this PR
</blockquote></details>

added
___
*[`260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java` line 2157 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOTCEJ2JicewIFx0Pts-r2-2157:-NOmYapN2rTlPqiGNm_d:b29yp0r) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java#L2157)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

we should have independent task types for each step
</blockquote></details>

Added `AwsLambdaDeployTask`
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTask.java` line 34 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOTCEJ2JicewIFx0Ptu-r2-34:-NOmZwsB3VcS-UP_gaQs:bkqrf2t) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/930-delegate-tasks/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambdaCommandTask.java#L34)):*
<details><summary><i>Previously, sainath-b (Sainath Batthala) wrote…</i></summary><blockquote>

@rohitk-h We shall use separate delegate task handler for each type of aws lambda task for backward compatibility
</blockquote></details>

agree, added `AwsLambdaDeployTask`
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambda.java` line 23 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOTCEJ2JicewIFx0Ptx-r2-23:-NOmj1Ay6K-tQTp3UDD3:b4rnasf) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/AwsLambda.java#L23)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

there should be tags , aliases and other fields as well. We can look into GetFunction Response DTO for the fields
</blockquote></details>

Can we add these as needed? Added tags, aliases are not being used yet. 
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaCommandRequest.java` line 61 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOjh5m-0ZdzXr3RCGyw:-NOmUvUI4BIjgI3xvqUp:b-bu91xm) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaCommandRequest.java#L61)):*
<details><summary><i>Previously, PragyeshMishra01 (Pragyesh Mishra) wrote…</i></summary><blockquote>

nit: We should add NPE check for awsLambdaArtifactConfig, if it is null for prepareRollbackRequest
</blockquote></details>

Updated
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaDeployRequest.java` line 36 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOTCEJ2JicewIFx0Pu3-r2-36:-NOmV38A3QeUU31ItoAm:b-etnly2) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaDeployRequest.java#L36)):*
<details><summary><i>Previously, piyush-bhuwalka wrote…</i></summary><blockquote>

@NonFinal @Expression(ALLOW_SECRETS) for functionName and qualifier as well
</blockquote></details>

good point, 
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaPrepareRollbackRequest.java` line 45 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOjfwWF43ka31YYkX0O:-NOmV7X6-0oSQJGZJhy0:b-skyv1y) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaPrepareRollbackRequest.java#L45)):*
<details><summary><i>Previously, PragyeshMishra01 (Pragyesh Mishra) wrote…</i></summary><blockquote>

is this intentional?
</blockquote></details>

Added todo, this a WIP, so will update in followup PR.

File: 950-delegate-tasks-beans/src/main/java/io/harness/serializer/kryo/DelegateTasksBeansKryoRegister.java [hunk 9]
```diff
@@ -2339,5 +2348,14 @@ public void register(Kryo kryo) {
     kryo.register(TerraformCloudRunTaskResponse.class, 680009);
 
     kryo.register(AwsSamInstallationCapability.class, 10000401);
+    kryo.register(AwsLambdaDeployRequest.class, 10000402);
```
Comment (sainath-b) [inline 950-delegate-tasks-beans/src/main/java/io/harness/serializer/kryo/DelegateTasksBeansKryoRegister.java:2351]: 100004XX series started for SAM
Please use different series for Lambda. May be 10000501

Comment (sainath-b) [reply 950-delegate-tasks-beans/src/main/java/io/harness/serializer/kryo/DelegateTasksBeansKryoRegister.java:2351]: @rohitk-h

File: 400-rest/src/test/resources/manager-kryo-registrations.txt [hunk 10]
```diff
@@ -3079,6 +3079,15 @@
 10000304:io.harness.delegate.beans.connector.terraformcloudconnector.TerraformCloudCredentialType
 
 10000401:io.harness.delegate.beans.executioncapability.AwsSamInstallationCapability
+10000402:io.harness.delegate.task.aws.lambda.request.AwsLambdaDeployRequest
+10000403:io.harness.delegate.task.aws.lambda.response.AwsLambdaDeployResponse
```
Comment (sainath-b) [inline 400-rest/src/test/resources/manager-kryo-registrations.txt:3083]: Please use different series for AWS Lambda. 100004XX is used for AWS SAM

File: 260-delegate/src/test/resources/kryo-registrations.txt [hunk 11]
```diff
@@ -2672,6 +2672,15 @@
 10000304:io.harness.delegate.beans.connector.terraformcloudconnector.TerraformCloudCredentialType
 
 10000401:io.harness.delegate.beans.executioncapability.AwsSamInstallationCapability
+10000402:io.harness.delegate.task.aws.lambda.request.AwsLambdaDeployRequest
```
Comment (sainath-b) [inline 260-delegate/src/test/resources/kryo-registrations.txt:2675]: Please use different series for AWS Lambda. 100004XX is used for AWS SAM

**Review by sainath-b — DISMISSED** [review]

**Review by rohitk-h — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44069)* status: 38 file(s) to review. 16. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): del-team. 

___
*[`260-delegate/src/test/resources/kryo-registrations.txt` line 2675 at r3](https://reviewable.io/reviews/harness/harness-core/44069#-NOmkifq2TfVJogn2dt3-r3-2675:-NOosoZz12PJtwGlZ9Km:bvnrpcm) ([raw file](https://github.com/harness/harness-core/blob/b36fa5c40448f492bce093c911bd4a3ce07a199a/260-delegate/src/test/resources/kryo-registrations.txt#L2675)):*
<details><summary><i>Previously, sainath-b (Sainath Batthala) wrote…</i></summary><blockquote>

Please use different series for AWS Lambda. 100004XX is used for AWS SAM
</blockquote></details>

updated
___
*[`400-rest/src/test/resources/manager-kryo-registrations.txt` line 3083 at r3](https://reviewable.io/reviews/harness/harness-core/44069#-NOmkifrUH6LXYdDHR27-r3-3083:-NOosmsh3t4e49i0iicx:boac9c8) ([raw file](https://github.com/harness/harness-core/blob/b36fa5c40448f492bce093c911bd4a3ce07a199a/400-rest/src/test/resources/manager-kryo-registrations.txt#L3083)):*
<details><summary><i>Previously, sainath-b (Sainath Batthala) wrote…</i></summary><blockquote>

Please use different series for AWS Lambda. 100004XX is used for AWS SAM
</blockquote></details>

updated. 
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/serializer/kryo/DelegateTasksBeansKryoRegister.java` line 2351 at r3](https://reviewable.io/reviews/harness/harness-core/44069#-NOTCEJ2JicewIFx0Pu6-r3-2351:-NOosvCV0_6uNq9pDTUA:bvnrpcm) ([raw file](https://github.com/harness/harness-core/blob/b36fa5c40448f492bce093c911bd4a3ce07a199a/950-delegate-tasks-beans/src/main/java/io/harness/serializer/kryo/DelegateTasksBeansKryoRegister.java#L2351)):*
<details><summary><i>Previously, sainath-b (Sainath Batthala) wrote…</i></summary><blockquote>

@rohitk-h 
</blockquote></details>

updated

**Comment by rohitk-h** [comment]
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaCommandRequest.java` line 61 at r2](https://reviewable.io/reviews/harness/harness-core/44069#-NOjh5m-0ZdzXr3RCGyw:-NOpCkCK7mWAxze8p7ZP:bnkqo4c) ([raw file](https://github.com/harness/harness-core/blob/5332906153c018ffacc98fc94ba30d37b12bde4c/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/aws/lambda/request/AwsLambdaCommandRequest.java#L61)):*
<details><summary><i>Previously, rohitk-h (Rohit Karelia) wrote…</i></summary><blockquote>

Updated
</blockquote></details>

As per PMD:
`No need to check for null before an instanceof; the instanceof keyword returns false when given a null argument.`

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — APPROVED** [review]
> Reviewed 10 of 10 files at r3, 3 of 3 files at r4, 10 of 10 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44069)* status: 46 file(s) to review. 11. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by piyush-bhuwalka — COMMENTED** [review]
> Reviewed 17 of 34 files at r1, 9 of 11 files at r2, 5 of 10 files at r3, 9 of 10 files at r5, 7 of 7 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44069)* status: 7 file(s) to review. 6. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 17 of 34 files at r1, 9 of 11 files at r2, 5 of 10 files at r3, 9 of 10 files at r5, 7 of 7 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44069)* status: 6. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by piyush-bhuwalka — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44069)* status: 6. unresolved discussions. Waiting on team(s): del-team.

**Review by mdmilic — APPROVED** [review]
approving only delegate owned files

> Reviewed 1 of 34 files at r1, 3 of 11 files at r2, 1 of 10 files at r5, 2 of 7 files at r6.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44069#-:-NOsoX5U5v04eVBr3o5z:bm3cxlm)* status: 6. unresolved discussions. Waiting on team(s): del-team.

**Review by gn-harness — APPROVED** [review]
> Reviewed 17 of 34 files at r1, 9 of 11 files at r2, 5 of 10 files at r3, 9 of 10 files at r5, 7 of 7 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44069)* status: 6. unresolved discussions.

**Review by piyush-bhuwalka — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44069)* status: :shipit: complete! ✓

**Pull request merged**